sqlglot/generator.py

```diff
@@ -1762,11 +1762,8 @@ def div_sql(self, expression: exp.Div) -> str:
 
     def floatdiv_sql(self, expression: exp.FloatDiv) -> str:
         if self.INTEGER_DIVISION:
-            # Multiply divident by 1.0 so that the expression's type can be coerced to float
-            this = exp.Mul(this=expression.this, expression=exp.Literal.number(1.0))
-            return self.div_sql(
-                exp.Div(this=exp.Paren(this=this), expression=expression.expression)
-            )
+            this = exp.Cast(this=expression.this, to=exp.DataType.build("DOUBLE"))
+            return self.div_sql(exp.Div(this=this, expression=expression.expression))
```

Comment:
this=exp.cast(expression.this, "double")

Reply:
Ah I forgot we added that, thanks.

Reply:
Discussed in slack: keeping it as is because for the cast to int right above this we can't use `exp.cast`, due to infinite recursion problems.